drivers/sensor/wsen_hids_2525020210002/wsen_hids_2525020210002.c

```diff
@@ -0,0 +1,255 @@
+		return -EFAULT;
+	}
+
+	switch ((int)attr) {
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_PRECISION:
+		val->val1 = data->sensor_precision;
+		val->val2 = 0;
+		break;
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_HEATER:
+		val->val1 = data->sensor_heater;
+		val->val2 = 0;
+		break;
+	default:
+		LOG_ERR("Operation not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api hids_2525020210002_driver_api = {
+	.attr_set = hids_2525020210002_attr_set,
+	.attr_get = hids_2525020210002_attr_get,
+	.sample_fetch = hids_2525020210002_sample_fetch,
+	.channel_get = hids_2525020210002_channel_get,
+};
+
+static int hids_2525020210002_init(const struct device *dev)
+{
+	const struct hids_2525020210002_config *const config = dev->config;
+	struct hids_2525020210002_data *data = dev->data;
+	struct sensor_value precision, heater;
+
+	/* Initialize WE sensor interface */
+	HIDS_Get_Default_Interface(&data->sensor_interface);
+	data->sensor_interface.interfaceType = WE_i2c;
+	data->sensor_interface.handle = (void *)&config->bus_cfg.i2c;
+
+	/* First communication test - check device ID */
+	if (HIDS_Sensor_Init(&data->sensor_interface) != WE_SUCCESS) {
+		LOG_ERR("Failed to read device ID.");
+		return -EIO;
+	}
+
+	precision.val1 = config->precision;
+	precision.val2 = 0;
+
+	if (hids_2525020210002_precision_set(dev, &precision) < 0) {
+		LOG_ERR("Failed to set precision configuration.");
+		return -EIO;
+	}
+
+	heater.val1 = config->heater;
+	heater.val2 = 0;
+
+	if (hids_2525020210002_heater_set(dev, &heater) < 0) {
+		LOG_ERR("Failed to set heater option.");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "HIDS-2525020210002 driver enabled without any devices"
+#endif
+
+/*
+ * Device creation macros
+ */
+
+#define HIDS_2525020210002_DEVICE_INIT(inst)                                                       \
+	SENSOR_DEVICE_DT_INST_DEFINE(inst, hids_2525020210002_init, NULL,                          \
+				     &hids_2525020210002_data_##inst,                              \
+				     &hids_2525020210002_config_##inst, POST_KERNEL,               \
+				     CONFIG_SENSOR_INIT_PRIORITY, &hids_2525020210002_driver_api);
+
+/*
+ * Main instantiation macro.
+ */
+#define HIDS_2525020210002_DEFINE(inst)                                                            \
+	static struct hids_2525020210002_data hids_2525020210002_data_##inst;                      \
+	static const struct hids_2525020210002_config hids_2525020210002_config_##inst = {         \
+		.bus_cfg = {.i2c = I2C_DT_SPEC_INST_GET(inst)},                                    \
+		.precision = (hids_2525020210002_precision_t)(DT_INST_ENUM_IDX(inst, precision)),  \
+		.heater = (hids_2525020210002_heater_t)(DT_INST_ENUM_IDX(inst, heater)),           \
+	};                                                                                         \
+	HIDS_2525020210002_DEVICE_INIT(inst)
```

Comment:
```suggestion
	SENSOR_DEVICE_DT_INST_DEFINE(inst, hids_2525020210002_init, NULL,                          \
				     &hids_2525020210002_data_##inst,                              \
				     &hids_2525020210002_config_##inst, POST_KERNEL,               \
				     CONFIG_SENSOR_INIT_PRIORITY, &hids_2525020210002_driver_api);
```

Reply:
this has been applied to all of the drivers.